libqtile/state.py

```diff
@@ -31,10 +32,17 @@ def __init__(self, qtile):
         # configurations.
         self.groups = {}
         self.screens = {}
+        self.focusHistory = {}
         self.current_screen = 0
+        self.layoutMap = {}
 
         for group in qtile.groups:
             self.groups[group.name] = group.layout.name
+            self.layoutMap[group.name] = {}
+            for layout in group.layouts:
+                self.layoutMap[group.name][layout.name] = layout
+                layout.group = None
```

Comment:
Why are you setting this to None? Why not use a group property for Layout? Is it set anywhere else?